src/lib/optimizer/strategy/predicate_pushdown_rule.cpp

```diff
@@ -0,0 +1,111 @@
+#include "predicate_pushdown_rule.hpp"
+#include "all_parameter_variant.hpp"
+#include "logical_query_plan/abstract_lqp_node.hpp"
+#include "logical_query_plan/join_node.hpp"
+#include "logical_query_plan/lqp_column_reference.hpp"
+#include "logical_query_plan/predicate_node.hpp"
+#include "logical_query_plan/sort_node.hpp"
+
+namespace opossum {
+
+std::string PredicatePushdownRule::name() const { return "Predicate Pushdown Rule"; }
+
+bool PredicatePushdownRule::apply_to(const std::shared_ptr<AbstractLQPNode>& node) {
+  // find predicate node
+  if (node->type() == LQPNodeType::Predicate) {
+    const auto parents = node->parents();
```

Comment:
We changed the LQP to have inputs and outputs (instead of children and parents). That should make some naming questions clearer.